#164691212 Integrate coveralls for test coverage reporting

richien opened the pull request:
#### What does this PR do?

Integrate coveralls to enable reporting on test coverage.

#### Description of Task to be completed?

-    Add coveralls token to the .travis.yml file
-    Add coveralls badge to README.md file
-    Add command for installation of coveralls by TravisCI in the travis.yml file
    by TravisCI

#### How should this be manually tested?

-   Click on the coverage badge in the screenshots section below.
-    This will take you to the coveralls site where you can view additional
    information about the coverage report.

#### Any background context you want to provide?

- The application is configured to use the Django-nose test suite to run the tests and coverage checks  are run when `python manage.py test` is run in the `script` section of the `.travis.yml` file. The coverage report is then sent to coveralls by a call to coveralls in `after-success` section of the `.travis.yml` file
 -  The coverage badge in the README file points to the develop branch and is expected to have a status [![Coverage Status](https://coveralls.io/repos/github/aqua-organisation/ah-django-Aqua/badge.svg?branch=develop)](https://coveralls.io/github/aqua-organisation/ah-django-Aqua?branch=develop) until this PR is merged into develop

What are the relevant pivotal tracker stories?

#164691212

###Screenshots

-    Below is the badge for the coverage on this branch
    [![Coverage Status](https://coveralls.io/repos/github/aqua-organisation/ah-django-Aqua/badge.svg?branch=ch-integrate-coveralls-164691212)](https://coveralls.io/github/aqua-organisation/ah-django-Aqua?branch=ch-integrate-coveralls-164691212)

RonKbS commented:
If you include a `wip` tag, it is the norm not to request for review until you remove it.
`wip` shows that you do not have the full implementation and so people looking to review PRs can hold off on reviewing one with the tag on it.

RonKbS commented:
Once a `wip` tag is removed, the practice is to not add it back after receiving reviews.
Kindly work on any changes requested without adding any more badges.

Regarding the changes introduced in this PR, I do not see how coveralls.io is to receive a report from which it will get the information to display in the badge.

richien commented:
> Once a `wip` tag is removed, the practice is to not add it back after receiving reviews.
> Kindly work on any changes requested without adding any more badges.
> 
> Regarding the changes introduced in this PR, I do not see how coveralls.io is to receive a report from which it will get the information to display in the badge.

Thank you for the feedback and I have taken note of it. Indeed the setup was incomplete as you had pointed out to me earlier. Thank you for your guidance.

reifred reviewed (approved):
Thanks for the work @richien. The coverage is now showing

NicholusMuwonge reviewed (approved):
This works fine. @richien good job

RonKbS commented on .travis.yml:
```diff
@@ -12,6 +12,9 @@ after_success:
   - postgresql
 
+before_install:
+- export COVERALLS_REPO_TOKEN=Y8eJAiu5n3BdD3BloaQULnqdZIywylTu3
```
A few recommendations;

- Please look up how to run coverage and report the results in coveralls by including commands in this file.
- This token may not be required but if you find it necessary, kindly look up a way of picking it from the environment rather than hardcoding it. It is the norm to not hard code variables that are exported to an environment.

richien replied:
Thank you for the recommendations. Let me get started on implementing them.

richien replied:
This has been implemented.

RonKbS replied:
The implementation is not yet complete. Particularly the first recommendation.

richien replied:
The application is configured to use the Django-nose test suite to run the tests which is why when the `python manage.py test` is run, the coverage is also run and displayed. The coverage report is then sent to coveralls by a call to `coveralls` in `after-success` section of the .travis.yml file

richien replied:
Kindly note, that I have added this to the `background context` section of the PR.

RonKbS reviewed (changes requested):
This implementation

richien committed:
chore(integrate coveralls): integrate coveralls code coverage

- add command instructing TravisCI to install coveralls in
the .travis.yml file. Coverage tests are already configured to run
 with the Django-nose test suite in the application settings.py file
- add the coveralls repo-token as an environment variable in the
repo settings on the travis site
- add the coveralls badge to the README.md file

[Delivers #164691212]
